src/App.tsx

```diff
@@ -1,36 +1,55 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState(moviesFromServer);
+
+  const filterMovies = (searchQuery: string) => {
+    const filteredMovies = moviesFromServer.filter(
+      (movie) => movie.title.toLowerCase()
+        .includes(searchQuery.trim().toLowerCase())
+        || movie.description.toLowerCase()
+          .includes(searchQuery.trim().toLowerCase()),
+    );
+
+    setVisibleMovies(filteredMovies);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setQuery(value);
+    filterMovies(value);
+  };
+
+  const memoizedVisibleMovies = useMemo(() => visibleMovies, [visibleMovies]);
+
   return (
     <div className="page">
       <div className="page-content">
         <div className="box">
           <div className="field">
-            {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
             <label htmlFor="search-query" className="label">
               Search movie
             </label>
-
             <div className="control">
               <input
                 type="text"
                 id="search-query"
                 className="input"
                 placeholder="Type search word"
+                value={query}
+                onChange={handleInputChange}
               />
             </div>
           </div>
         </div>
-
-        <MoviesList movies={moviesFromServer} />
-      </div>
-
-      <div className="sidebar">
-        Sidebar goes here
+        <MoviesList movies={memoizedVisibleMovies} />
```

Comment:
you can use just result of your filterMovies function 
`memoizedVisibleMovies` -> `filterMovies(query)`